### PR TITLE
how-to, toc: add TiDB in K8s troubleshooting

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? 

Add the TiDB in Kubernetes troubleshooting document.

> **Note:** This PR contains a link that points to the local PV document which hasn't been added to TOC.md yet. I'll add it soon. This PR can be merged after that file is added to TOC.

### What is the related PR or file link(s)? 

- https://github.com/pingcap/docs-cn/pull/1439
- https://github.com/pingcap/docs-cn/pull/1560
- https://github.com/pingcap/docs-cn/pull/1601

### Which version does your change affect? 

dev, v3.0

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+```shell
+kubectl get tidbclusters -n ${namespace}
+kubectl get statefulsets -n ${namespace}
+kubectl describe statefulsets -n ${namespace} ${cluster-name}-pd
+```
+
+## The network connection issue between Pods
+
+In a TiDB cluster, you can access most Pods by using the Pod's domain name (allocated by the Headless Service). The exception is when TiDB Operator collects the cluster information or issues the control command, it accesses the PD (Placement Driver) cluster using the `ServiceName` of the PD service.
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
In a TiDB cluster, you can access most Pods by using the Pod's domain name (allocated by the Headless Service). The exception is when TiDB Operator collects the cluster information or issues control commands, it accesses the PD (Placement Driver) cluster using the `ServiceName` of the PD service.
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+When you find some network connection issue exists between Pods from the log or monitoring metrics, or you find the network connection between Pods might be abnormal according to the problem condition, you can follow the following process to diagnose and gradually narrow down the problem:
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
When you find some network connection issues between Pods from the log or monitoring metrics, or you find the network connection between Pods might be abnormal according to the problematic condition, you can follow the following process to diagnose and narrow down the problem:
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+    After the remote shell is started, use the `dig` command to diagnose the DNS resolution. If the DNS resolution is abnormal, refer to [Debugging DNS Resolution](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/) for troubleshooting.
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    dig ${HOSTNAME}
+    ```
+
+    Use the `ping` command to diagnose whether the three layers of network to the destination IP are connected (the destination IP is the ClusterIP resolved using `dig`):
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
    Use the `ping` command to diagnose the connection with the destination IP (the ClusterIP resolved using `dig`):
```

Comment (dcalvin) [reply dev/how-to/troubleshoot/tidb-in-kubernetes.md]: Ping command sends ICMP packets which should belong to the layer 3 network, but I don't think we need to mention it.

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/pd/pd.toml | grep client-urls
+        # The default port of PD is 20160 when not configured.
+        kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/tikv/tikv.toml | grep addr
+        # The default port of TiDB is 4000 when not configured.
+        kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/tidb/tidb.toml | grep port
+        ```
+
+## The Pod is in the Pending state
+
+The Pending state of a Pod is usually caused by insufficient resources, such as the following conditions:
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
The Pending state of a Pod is usually caused by conditions of insufficient resources, such as:
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+You can check the specific reason for Pending by using the `kubectl describe pod` command:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl describe po -n ${namespace} ${pod_name}
+```
+
+- If the CPU or memory resources are insufficient, you can lower the CPU or memory resources applied by the corresponding component, or add a new Kubernetes node.
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
- If the CPU or memory resources are insufficient, you can lower the CPU or memory resources requested by the corresponding component for scheduling, or add a new Kubernetes node.
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+- If the `StorageClass` of the PVC cannot be found, delete the TiDB Pod and the corresponding PVC. Then, in the `values.yaml` file, change `StorageClassName` to the name of the `StorageClass` available in the cluster. Run the following command to get the `StorageClass` available in the cluster:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get storageclass
+    ```
+
+- If a `StorageClass` exists in the cluster but the available PV is insufficient, you need to add the corresponding PV resources. For Local PV, you can expand it by referring to [Local PV Configuration](/reference/configuration/tidb-in-kubernetes/local-pv-configuration.md).
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
- If a `StorageClass` exists in the cluster but the available PV is insufficient, you need to add PV resources correspondingly. For Local PV, you can expand it by referring to [Local PV Configuration](/reference/configuration/tidb-in-kubernetes/local-pv-configuration.md).
```

Comment (dcalvin) [reply dev/how-to/troubleshoot/tidb-in-kubernetes.md]: Cannot find the reference object in the context by saying "the corresponding PV resources".
But we can use the adv for the verb "add".

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+## The Pod is in the `CrashLoopBackOff` state
+
+A Pod in the `CrashLoopBackOff` state means that the container in the Pod repeatedly exits abnormally. After one abnormal exit, the container is restarted by kubelet, and then it exits abnormally again after the restart. Many reasons can cause `CrashLoopBackOff`. In this case, the most effective way to locate it is to view the log of the Pod container:
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
A Pod in the `CrashLoopBackOff` state means that the container in the Pod repeatedly aborts, in the loop of abort - restart by `kubelet1 - abort. There are many potential causes of `CrashLoopBackOff`. In this case, the most effective way to locate it is to view the log of the Pod container:
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+In addition, TiKV might also fail to start when `ulimit` is insufficient. In this case, you can modify the `/etc/security/limits.conf` file of the Kubernetes node to increase the `ulimit`:
+
+```
+root soft nofile 1000000
+root hard nofile 1000000
+root soft core unlimited
+root soft stack 10240
+```
+
+If you cannot confirm the cause from the log and `ulimit` is also a normal value, troubleshoot it by using [the diagnostic mode](#use-the-diagnostic-mode).
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
If you cannot confirm the cause from the log and `ulimit` is also a normal value, troubleshoot it further by using [the diagnostic mode](#use-the-diagnostic-mode).
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+
+## Unable to access the TiDB service
+
+If you cannot access the TiDB service, first check whether the TiDB service is deployed successfully using the following approach:
```
Comment (dcalvin) [inline dev/how-to/troubleshoot/tidb-in-kubernetes.md]:
```suggestion
If you cannot access the TiDB service, first check whether the TiDB service is deployed successfully using the following method:
```

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get tidbclusters -n ${namespace}
+kubectl get statefulsets -n ${namespace}
+kubectl describe statefulsets -n ${namespace} ${cluster-name}-pd
+```
+
+## The network connection issue between Pods
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/how-to/troubleshoot/tidb-in-kubernetes.md]: Connection issue maybe a little wider, and "the" is not necessary here.

[the remark this answers is not in the document]
Comment (lilin90) [reply dev/how-to/troubleshoot/tidb-in-kubernetes.md]: Agree with you 😀

File: dev/how-to/troubleshoot/tidb-in-kubernetes.md (hunk 7 above)
Comment (lilin90) [reply dev/how-to/troubleshoot/tidb-in-kubernetes.md]: You must mean `kubelet` by "`kubelet1".

**Pull request merged**